analytics/src/main/java/com/segment/analytics/internal/Utils.java

```diff
@@ -532,6 +533,16 @@ public static void copySharedPreferences(SharedPreferences src, SharedPreference
         editor.apply();
     }
 
+    private static Context obtainDeviceProtectedStorageContext(Context context) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            // All N devices have split storage areas, but we may need to recreate the preferences in the new device
+            // protected storage area, which is where the data lives from now on.
+            return context.createDeviceProtectedStorageContext();
```

Comment:
Would it be a backward-compatibility issue? Not 100% sure but probably it's the client job to specify which context should be used to access the storage area - device or user...

Perhaps we can do better - we can extend the `Builder` [class](https://github.com/segmentio/analytics-android/blob/master/analytics/src/main/java/com/segment/analytics/Analytics.java#L1061)

Reply:
good callout. once we get the test plan i will try to run it via multiple scenarios to make sure nothing breaks 👍

Reply:
I don't think it is client responsibility think of which context can be specified, given that any needed check can be done on the library side and also any type of context can be obtained.
Context class itself has those api to get application context or to get context with device protected storage reference. It just doesn't make sense to burden clients with all those decisions.

Reply:
I'll just move prefs to device protected storage as a migration step.

Reply:
> I don't think it is client responsibility think of which context can be specified

Yeah, may be it's not the client responsibility... It mostly depends on what type of data the segment library keeps in the shared preferences.

According to https://android-developers.googleblog.com/2016/04/developing-for-direct-boot.html
> All sensitive, private information, like the full message history and a read/write access token, should still be saved in credential protected storage.

Regarding
> I'll just move prefs to device protected storage as a migration step.

This's going to be tricky since we should take into account that the credential protected storage can be unaccessible (yet).